### PR TITLE
generate_makefile: allow enabling of hwloc and memkind tpls

### PR DESCRIPTION
**Pull request by ndellingwood**

### CODE REVIEW

File: generate_makefile.bash [hunk 1]
```diff
@@ -301,6 +303,18 @@ else
     KOKKOS_DEBUG_CMD=-DCMAKE_BUILD_TYPE=DEBUG
 fi
 
+if [ "$KOKKOS_HWLOC" == "" ]; then
```
Comment (dalg24) [inline generate_makefile.bash]: That's a bit silly.  If I change the definition above `ON -> OFF` and don't pay attention I will be surprised.
(Not requesting changes, just a comment.)

Comment (ndellingwood) [reply generate_makefile.bash]: Yeah, not sure why I added it this way, tired at the end of the day I guess. I'll change it.

Comment (crtrott) [reply generate_makefile.bash]: Change ot to checking for ON and make the empty declaration in the else clause.

**Comment by ndellingwood** [comment]
Ready for merge.

**Pull request merged**